types/consensus_data.go

```diff
@@ -156,36 +147,30 @@ func (ci *ConsensusData) GetAttestationData() (*phase0.AttestationData, error) {
 	return ret, nil
 }
 
-func (ci *ConsensusData) GetBellatrixBlockData() (*bellatrix.BeaconBlock, error) {
-	ret := &bellatrix.BeaconBlock{}
-	if err := ret.UnmarshalSSZ(ci.DataSSZ); err != nil {
-		return nil, errors.Wrap(err, "could not unmarshal ssz")
-	}
-	return ret, nil
-}
-
-func (ci *ConsensusData) GetBellatrixBlindedBlockData() (*bellatrix2.BlindedBeaconBlock, error) {
-	ret := &bellatrix2.BlindedBeaconBlock{}
-	if err := ret.UnmarshalSSZ(ci.DataSSZ); err != nil {
-		return nil, errors.Wrap(err, "could not unmarshal ssz")
-	}
-	return ret, nil
-}
-
-func (ci *ConsensusData) GetCapellaBlockData() (*capella.BeaconBlock, error) {
-	ret := &capella.BeaconBlock{}
-	if err := ret.UnmarshalSSZ(ci.DataSSZ); err != nil {
-		return nil, errors.Wrap(err, "could not unmarshal ssz")
+func (ci *ConsensusData) GetBlockData() (*spec.VersionedBeaconBlock, ssz.HashRoot, error) {
+	switch ci.Version {
```

Comment:
if we're basically deprecating Bellatrix, this is a preparation to support future version ?

Reply:
exactly, so if for example next fork [`Deneb`](https://crypto.news/days-after-shanghai-ethereum-developers-readying-for-key-eip-and-deneb-upgrade/) will introduce new block, we just need to upgrade the eth2-go-client package and adjust the code to use the `Deneb` beacon block